api/handler/v1beta1/job_spec.go

```diff
@@ -163,30 +163,53 @@ func (sv *JobSpecServiceServer) CreateJobSpecification(ctx context.Context, req
 		return nil, mapToGRPCErr(sv.l, err, "unable to get namespace")
 	}
 
-	jobSpec, err := FromJobProto(req.GetSpec(), sv.pluginRepo)
+	reqJobSpec := req.GetSpec()
+	jobSpec, err := FromJobProto(reqJobSpec, sv.pluginRepo)
 	if err != nil {
 		return nil, status.Errorf(codes.Internal, "cannot deserialize job: \n%s", err.Error())
 	}
+	jobSpecs := []models.JobSpec{jobSpec}
 
-	// validate job spec
-	if err = sv.jobSvc.Check(ctx, namespaceSpec, []models.JobSpec{jobSpec}, logWriter); err != nil {
-		return nil, status.Errorf(codes.Internal, "spec validation failed\n%s", err.Error())
+	deployID, err := sv.jobSvc.CreateAndDeploy(ctx, namespaceSpec, jobSpecs, logWriter)
+
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "job addition failed for project %s, error:: %s", req.GetProjectName(), err.Error())
 	}
+	runtimeDeployJobSpecificationCounter.Inc()
+
+	return &pb.CreateJobSpecificationResponse{
+		Success: true,
+		Message: fmt.Sprintf("job is created and queued for deployment on project %s, with DeploymentId :: %s", req.GetProjectName(), deployID.UUID().String()),
```

Comment:
- looks like the colon is doubled
- lets change `DeploymentID` to `Deployment ID`